### ISSUE TITLE
1.8.1 breaks solid start server side redirects

### ISSUE THREAD

**Issue by apatrida** [solidjs/solid#1922]
### Describe the bug

Since 1.8.1 solidjs, breaks server redirects in solid start.

in route data:

```
export function homeData() {
  return createServerData$(async (_, { locals }) => {
    const unsafeSession = useSession$(locals);
    if (!unsafeSession.isAuthed) {
      throw redirect("...");
    } else {
      throw redirect("...);
    }
  });
}
```

goes to Seroval for some reason which then blows up with error:

```
An unhandled error occured: Error: Unsupported type "[object Response]"
    at Gt (file:///.../node_modules/.pnpm/seroval@0.10.4/node_modules/seroval/dist/esm/production/index.mjs:3:43393)
    at R (file:///.../node_modules/.pnpm/seroval@0.10.4/node_modules/seroval/dist/esm/production/index.mjs:3:43584)
    at n.then.e.alive.e.onParse.t (file:///.../node_modules/.pnpm/seroval@0.10.4/node_modules/seroval/dist/esm/production/index.mjs:3:40175)
```

### Your Example Website or App

sample included above

### Steps to Reproduce the Bug or Issue

do a server side redirect in route data with `throw redirect("...url...")`

### Expected behavior

a redirect, not an exception

### Platform

all

**Comment by ryansolid** [comment]
Turns out this is a bug in SolidStart. I that just was sort of slipping by. I will update and do a release.

**Comment by ryansolid** [comment]
Ok this should be fixed 0.3.7 of SolidStart.

**Comment by apatrida** [comment]
thanks!!
